File: cpp/Gisketch/Aottg2UI/Game/AottgProfileEntitlementOption.c
// Type: Gisketch.Aottg2UI.Game.AottgProfileEntitlementOption
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Gisketch.Aottg2UI.Game/AottgProfileEntitlementOption.cs
// Prior real C# source: none
// --------------------------------

// Gisketch.Aottg2UI.Game.AottgProfileEntitlementOption$$.ctor
// il2cpp: void Gisketch_Aottg2UI_Game_AottgProfileEntitlementOption___ctor (Gisketch_Aottg2UI_Game_AottgProfileEntitlementOption_o __this, System_String_o* id, System_String_o* label, System_String_o* icon, bool enabled, const MethodInfo* method);
// 0x4526b20

void Gisketch_Aottg2UI_Game_AottgProfileEntitlementOption___ctor
               (Gisketch_Aottg2UI_Game_AottgProfileEntitlementOption_o __this,System_String_o *id,
               System_String_o *label,System_String_o *icon,bool_conflict enabled,MethodInfo *method)

{
  System_String_Fields SVar1;
  undefined4 in_register_0000000c;
  
  SVar1._4_4_ = in_register_0000000c;
  SVar1._stringLength = enabled;
  id->klass = (System_String_c *)label;
  il2cpp_runtime_helper_022b4080();
  id->monitor = icon;
  il2cpp_runtime_helper_022b4080(&id->monitor,icon);
  id->fields = SVar1;
  il2cpp_runtime_helper_022b4080(&id->fields,SVar1);
  *(char *)&id[1].klass = (char)method;
  return;
}

File: csharp_body_deref/Scripts/Gisketch.Aottg2UI.Game/AottgProfileEntitlementOption.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Gisketch.Aottg2UI.Game.AottgProfileEntitlementOption
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Gisketch/Aottg2UI/Game/AottgProfileEntitlementOption.c
// Prior real C# source: none (type is newer than the last public source snapshot)
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace Gisketch.Aottg2UI.Game;

[Token(Token = "0x20001B8")]
internal readonly struct AottgProfileEntitlementOption
{
	[Token(Token = "0x4000969")]
	[FieldOffset(Offset = "0x0")]
	public readonly string Id;

	[Token(Token = "0x400096A")]
	[FieldOffset(Offset = "0x8")]
	public readonly string Label;

	[Token(Token = "0x400096B")]
	[FieldOffset(Offset = "0x10")]
	public readonly string Icon;

	[Token(Token = "0x400096C")]
	[FieldOffset(Offset = "0x18")]
	public readonly bool Enabled;

	[Token(Token = "0x6000A7B")]
	[Address(RVA = "0x4426B20", Offset = "0x4426B20", VA = "0x4426B20")]
	public AottgProfileEntitlementOption(string id, string label, string icon, bool enabled = true)
	{
	}
}
